Source/OnboardingContentViewController.m

```diff
@@ -29,10 +30,17 @@
 
 @interface OnboardingContentViewController ()
 
+@property (nonatomic, strong) MPMoviePlayerController *moviePlayerController;
+@property (nonatomic, strong) NSURL *videoURL;
+
 @end
 
 @implementation OnboardingContentViewController
 
+- (void)dealloc {
+    [self removeObserver:self forKeyPath:UIApplicationWillEnterForegroundNotification];
```

Comment:
Kinda similar, ```[[NSNotificationCenter defaultCenter] removeObserver:self]``` will remove all the observers and this could potentially remove a observer you don't want to, let's say you call it in ```viewWillDisappear:``` (I know, this is a worst case scenario as ```removeObserver:``` shouldn't be used outside of ```dealloc``` in that case it is not really that bad) it removes all the observers (even the ones in its superclass) and maybe the superclass need them at some point as it is still alive and you know, that make kittens sad, with ```[[NSNotificationCenter defaultCenter] removeObserver: keyPath:]``` you are specifying the observer you want to remove and won't have problems with the others, I use it as a good practice.